src/SimpleWallet/SimpleWallet.cpp

```diff
@@ -1954,6 +1956,114 @@ bool simple_wallet::save_keys_to_file(const std::vector<std::string>& args)
   return true;
 }
 
+bool simple_wallet::save_all_txs_to_file(const std::vector<std::string> &args)
+{
+  if (!args.empty())
+  {
+    logger(ERROR) <<  "Usage: \"save_txs_to_file\"";
+    return true;
+  }
+
+  /* get total txs in wallet */
+  size_t tx_count = m_wallet->getTransactionCount();
+
+  /* check wallet txs before doing work */
+  if (tx_count == 0) {
+    logger(ERROR, BRIGHT_RED) << "No transfers";
+    return true;
+  }
+
+  // we do should do optional bool to include deposits
+
+  /* remove ".wallet" from the end of the string to create filename */
+  std::string formatted_wal_str = m_wallet_file.erase(m_wallet_file.size() - 7) + "_conceal_transactions.txt";
+  std::ofstream tx_file(formatted_wal_str);
+
+  /* create line from string */
+  std::string listed_tx;
+
+  /* get tx struct */
+  WalletLegacyTransaction txInfo;
+
+  /* create header for listed txs */
+  std::string header = common::makeCenteredString(32, "timestamp (UTC)") + "  ";
+  header += common::makeCenteredString(64, "hash") + "  ";
+  header += common::makeCenteredString(20, "total amount") + "  ";
+  header += common::makeCenteredString(14, "fee") + "  ";
+  header += common::makeCenteredString(8, "block") + "  ";
+  header += common::makeCenteredString(12, "unlock time");
+
+  /* make header string to ss so we can use .size() */
+  std::stringstream hs(header + "\n");
+  hs << std::string(header.size(), '-');
+
+  /* push header to start of file */
+  tx_file << hs.str();
+
+  for (size_t i = 0; i < tx_count; ++i) 
+  {
+    /* get tx to list from i */
+    m_wallet->getTransaction(i, txInfo);
+
+    /* check tx state */
+    if (txInfo.state != WalletLegacyTransactionState::Active || txInfo.blockHeight == WALLET_LEGACY_UNCONFIRMED_TRANSACTION_HEIGHT) {
+      continue;
+    }
+
+    /* grab tx info */
+    std::string formatted_item = list_tx_item(txInfo, listed_tx);
+
+    /* push info to end of file */
+    tx_file << formatted_item;
+  }
+
+  return true;
+}
+
+std::string simple_wallet::list_tx_item(const WalletLegacyTransaction& txInfo, std::string listed_tx)
+{
+  std::vector<uint8_t> extraVec = common::asBinaryArray(txInfo.extra);
+
+  crypto::Hash paymentId;
+  std::string paymentIdStr = (getPaymentIdFromTxExtra(extraVec, paymentId) && paymentId != NULL_HASH ? common::podToHex(paymentId) : "");
+
+  char timeString[32 + 1];
+  time_t timestamp = static_cast<time_t>(txInfo.timestamp);
+  struct tm time;
+#ifdef _WIN32
+  gmtime_s(&time, &timestamp);
+#else
+  gmtime_r(&timestamp, &time);
+#endif
+
+  if (!std::strftime(timeString, sizeof(timeString), "%c", &time))
+  {
+    throw std::runtime_error("time buffer is too small");
+  }
+
+  std::string format_amount = m_currency.formatAmount(txInfo.totalAmount);
+  std::string is_minus = txInfo.totalAmount < 0 ? "-" + format_amount : "" + format_amount;
```

Comment:
Not needed, this will add one minus to the formatted amount already containing one